client/core/helpers.go

```diff
@@ -310,29 +310,38 @@ func (ord *OrderReader) StatusString() string {
 
 // SimpleRateString is the formatted match rate.
 func (ord *OrderReader) SimpleRateString() string {
+	if ord.Type == order.MarketOrderType {
+		return ord.AverageRateString()
+	}
 	return ord.formatRate(ord.Rate)
 }
 
 // RateString is a formatted rate with units.
 func (ord *OrderReader) RateString() string {
+	rateStr := ord.formatRate(ord.Rate)
 	if ord.Type == order.MarketOrderType {
-		return "market"
+		rateStr = ord.AverageRateString()
 	}
-	return fmt.Sprintf("%s %s/%s", ord.formatRate(ord.Rate), ord.QuoteSymbol, ord.BaseSymbol)
+	return fmt.Sprintf("%s %s/%s", rateStr, ord.QuoteSymbol, ord.BaseSymbol)
 }
 
 // AverageRateString returns a formatting string containing the average rate of
 // the matches that have been filled in an order.
 func (ord *OrderReader) AverageRateString() string {
-	if len(ord.Matches) == 0 {
+	nMatch := len(ord.Matches)
+	if nMatch == 0 {
 		return "0"
 	}
 	var baseQty, rateProduct uint64
 	for _, match := range ord.Matches {
 		baseQty += match.Qty
 		rateProduct += match.Rate * match.Qty // order ~ 1e16
 	}
-	return ord.formatRate(rateProduct / baseQty)
+	rateStr := ord.formatRate(rateProduct / baseQty)
+	if nMatch > 1 {
+		rateStr = "~ " + rateStr // "~" only makes sense if the order has more than one match.
```

Comment:
Actually, `AverageRateString` is used in CSV files too, so can't actually do this here at all. I think you need to confine all changes to `(*OrderReader).RateString`.

Reply:
I don't see the "~" in the exported CSV.

Reply:
I think already fixed @buck54321 concern before you tested.